tests/tests/todo/bitvec_const02.rs

```diff
@@ -0,0 +1,27 @@
+#[flux::opaque]
+#[flux::refined_by(val: bitvec<32>)]
+#[derive(PartialEq, Eq)]
+pub struct BV32(u32);
+
+impl BV32 {
+    #[flux::trusted]
+    #[flux::sig(fn (x:u32) -> BV32[bv_int_to_bv32(x)])]
+    const fn new(val: u32) -> Self {
+        BV32(val)
+    }
+}
+
+pub trait Foo {
+    #[flux_rs::constant(bv_int_to_bv32(0x4567))]
+    const START: BV32 = BV32::new(0x4567);
```

Comment:
My point is that the behavior you get from this annotation will be wrong if not done carefully because you can override the const in an implementation

```rust
impl Foo for Sometype {
    const START: BV32 = BV32::new(0x1234);
 }
 ```
 
 So the usual trait shenanigans apply, i.e., in the trait definition we need to treat the const abstractly and then find the concrete one by normalizing projections.
 
 I think we can make sense of the annotation by considering it only for implementations that don't override the definition (i.e., they use the default)

Reply:
yup indeed.